src/main/java/com/mongodb/kafka/connect/sink/NoStackTraceDlqException.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mongodb.kafka.connect.sink;
+
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.sink.ErrantRecordReporter;
+
+import com.mongodb.lang.Nullable;
+
+/**
+ * An exception that is meant to be reported to the DLQ via {@link ErrantRecordReporter}. The
+ * {@linkplain Class#getName() class name} and {@linkplain #getMessage() message} must be treated as
+ * public API. This is necessary to allow the possibility of processing the DLQ in an automated way.
+ * The length of the {@linkplain #getStackTrace() stack trace} is always 0, which means that this
+ * exception is suitable for reporting only situations with the stack trace being inconsequential.
+ */
+abstract class NoStackTraceDlqException extends DataException {
+  private static final long serialVersionUID = 1L;
```

Comment:
Suggestion add an abstract version field - So all versions of NoStackTraceDlqExceptions have to implement their own version number

Reply:
`serialVersionUID` must be a `static` field, there is no way to express it as an `abstract` method, because its value is a characteristic of a class, not that of an object. Could you please clarify the suggestion?

P.S. Note also that `ErrantRecordReporter.report` does not use the Java serialization mechanism when reporting exceptions.

Reply:
Apologies - crossed wires here I was asking to consider adding a version field for each DLQ Exception. We've discussed this else where - so resolving this.